website/content/docs/api-gateway/upgrade-specific-versions.mdx

```diff
@@ -0,0 +1,159 @@
+that explicitly allows traffic from the routes namespace to the backendrefs namespace.
+
+
+
+## Requirements
+
+- The consul-api-gateway should be running version v0.1.0, if the version is different, follow the normal upgrade path
+- [jq](https://stedolan.github.io/jq/download/) is installed on the users CLI
+
+Note, as of writing, this is the only upgrade path documented as v0.1.0 is the only currently released version. Normal upgrade path will be documented
+in future releases
+
+## Assumptions
+
+This guides makes the following assumptions:
+
+- You have the ability to run Kubectl cli commands
+- Your kubectl config is already configured to point at the cluster with the installation you are upgrading
+- You have HTTPRoute.read, TCPRoute.read and ReferencePolicy.create rights on your kubernetes cluster
+
+
+## Procedures
+
+**1.** Optional, double check the current version of the consul-api-gateway with the following command
+
+```
+kubectl get deployment --namespace consul consul-api-gateway-controller -o json | jq '.spec.template.spec.containers[] | select(.name="api-gateway-controller") .image'
+```
+
+If the output looks as follows
+
+```
+"hashicorp/consul-api-gateway:0.1.0"
+```
+
+ continue following this upgrade path.
+
+**2.** Run the following command to retrieve all routes defined with a cross namespace:
+
+```
+kubectl get HttpRoute -o json -A | jq -r '.items[] | {name: .metadata.name, namespace: .metadata.namespace, kind: .kind, crossNamespaceBackendReferences: ( .metadata.namespace as $parentnamespace | .spec.rules[] .backendRefs[] | select(.namespace != null and .namespace != $parentnamespace )  )}  ' && kubectl get TcpRoute -o json -A | jq -r '.items[] | {name: .metadata.name, namespace: .metadata.namespace, kind: .kind, crossNamespaceBackendReferences: ( .metadata.namespace as $parentnamespace | .spec.rules[] .backendRefs[] | select(.namespace != null and .namespace != $parentnamespace )  )}  '
+
+```
+
+Note, the above command will retrieve all HTTPRoutes and TCPRoutes. TLSRoutes and UDPRoutes are not supported in v0.1.0
+
+If you have any breaking namespaces, your output will something like as follows. If your output is empty, you have no breaking routes and you can skip to the last step.
+
+```
+{
+  "name": "example-breaking-http-route",
+  "namespace": "example-namespace",
+  "kind": "HTTPRoute",
+  "crossNamespaceBackendReferences": {
+    "group": "",
+    "kind": "Service",
+    "name": "web-backend",
+    "namespace": "gateway-namespace",
+    "port": 8080,
+    "weight": 1
+  }
+}
+{
+  "name": "example-breaking-tcp-route",
+  "namespace": "a-different-namespace",
+  "kind": "TCPRoute",
+  "crossNamespaceBackendReferences": {
+    "group": "",
+    "kind": "Service",
+    "name": "web-backend",
+    "namespace": "gateway-namespace",
+    "port": 8080,
+    "weight": 1
+  }
+}
+```
+
+**3.** Create a ReferencePolicy to allow cross namespace traffic.
+
```

Comment:
We need to explain that a ReferencePolicy needs to be created for each pair of route/service namespaces.